snuba/query/parser/__init__.py

```diff
@@ -386,9 +396,10 @@ def visit_subscriptable_reference(self, exp: SubscriptableReference) -> Expressi
                 self.__expand_nested,
             )
         )
-        assert isinstance(
-            expanded_column, Column
-        ), "A subscriptable column cannot be resolved to anything other than a column"
```

Comment:
It seems like this would arise out of an issue with the dataset/visitor definition, and not anything that the outside caller could control. I think an assertion (or some other sort of non-user facing validation error) might be okay here?